### DISCUSSION TITLE
Input.Search的enterButton属性能不能控制搜索按钮是否显示?

### DISCUSSION THREAD

**Discussion by fnoopv** [ant-design/ant-design#41426] [Q&A]
[官网文档](https://ant.design/components/input-cn#inputsearch)文档中的说明是 `是否有确认按钮，可设为按钮文字。该属性会与 addonAfter 冲突。`, 类型为 `boolean | ReactNode`, 默认为值`false`，该怎么设置才能隐藏搜索框后面的按钮？我设置为`false` `null` `undefined`都不行

**Comment by afc163** [comment]
不需要搜索按钮的话，直接用 Input 就好？

**Reply by Yuiai01** [reply]
> 不需要搜索按钮的话，直接用 Input 就好？

+1，感觉还可以选择 select 组件的 `showSearch`，可见 https://codesandbox.io/s/sou-suo-kuang-antd-5-3-2-jw39xf?from-embed

**Reply by fnoopv** [reply]
是可以，我只是疑惑这个enterButton的实际作用和文档里面的说明是不是有点不符

**Reply by Yuiai01** [reply]
> 是可以，我只是疑惑这个enterButton的实际作用和文档里面的说明是不是有点不符

确实，由于 Input.Search 本身存在搜索确认按钮，感觉 `是否有确认按钮` 这句话会让人觉得可以控制按钮是否存在 🤔

**Reply by acyza** [reply]
可能是旧版本enterButton=false时，只是一个搜索图标，并不是正儿八经的按钮，所以有了这么一段说明 https://3x.ant.design/components/input-cn/#inputsearch
